### PR TITLE
Use property classes in single table QualityReport

### LINKED ISSUE #370: Use property classes in single table QualityReport
### Problem Description

Once #355 and #356 are complete, we should use the property classes in the `QualityReport` itself.

### Expected behavior

- The `get_details` method should just return the details from the desired property class.
- The `get_visualization` should just be a wrapper for the property class' `get_visualization`
- The `generate` method should loop over the properties and use a class for each one to get the property scores and then average in the end.

### Additional context

- The `get_raw_result` method can be removed

### PR DESCRIPTION
**Pull request by R-Palazzo**
Resolve #370.

### CODE REVIEW

File: sdmetrics/reports/single_table/quality_report.py [hunk 1]
```diff
@@ -25,35 +18,71 @@ class QualityReport():
...
+        metadata_columns = set(metadata['columns'].keys())
+
+        missing_data = metadata_columns.difference(real_columns.union(synthetic_columns))
+        missing_metadata = real_columns.union(synthetic_columns).difference(metadata_columns)
+        missing_columns = missing_data.union(missing_metadata)
+
+        if missing_columns:
+            error_message = (
+                'The metadata does not match the data. The following columns are missing'
+                ' in the real/synnthetic data or in the metadata: '
```
Comment (amontanez24) [inline sdmetrics/reports/single_table/quality_report.py:52]: typo: synnthetic

File: sdmetrics/reports/single_table/quality_report.py [hunk 2]
```diff
@@ -68,42 +97,33 @@ def generate(self, real_data, synthetic_data, metadata, verbose=True):
...
+            )
 
-        self._overall_quality_score = np.nanmean(list(self._property_breakdown.values()))
+        self._overall_quality_score = np.nanmean(scores)
+        self.is_generated = True
 
         if verbose:
             self._print_results()
 
+    def _validate_property_generation(self, property_name):
```
Comment (amontanez24) [inline sdmetrics/reports/single_table/quality_report.py:114]: minor: I'd rename to `_validate_property_generated`

File: sdmetrics/reports/single_table/quality_report.py [hunk 3]
```diff
@@ -25,35 +18,71 @@ class QualityReport():
...
+
+        Args:
+            real_data (pandas.DataFrame):
+                The real data.
+            synthetic_data (pandas.DataFrame):
+                The synthetic data.
+            metadata (dict):
+                The metadata of the table.
+        """
+        missing_columns = set()
```
Comment (amontanez24) [inline sdmetrics/reports/single_table/quality_report.py:40]: I don't think you need to define this here

File: tests/integration/reports/single_table/test_single_table_quality_report.py [hunk 4]
```diff
@@ -1,59 +1,273 @@
...
+        expected_details_cpt = pd.DataFrame(expected_details_cpt__dict)
+
+        pd.testing.assert_frame_equal(
+            report.get_details('Column Shapes'), expected_details_column_shapes
+        )
+        pd.testing.assert_frame_equal(
+            report.get_details('Column Pair Trends'), expected_details_cpt
+        )
+
+    def test_report_end_to_end_with_errors(self):
```
Comment (amontanez24) [inline tests/integration/reports/single_table/test_single_table_quality_report.py:121]: minor: missing docstring

File: tests/integration/reports/single_table/test_single_table_quality_report.py [hunk 5]
```diff
@@ -1,59 +1,273 @@
...
+        report.generate(real_data, synthetic_data, metadata)
+        properties = report.get_properties()
+
+        # Assert
+        pd.testing.assert_frame_equal(properties, pd.DataFrame({
+            'Property': ['Column Shapes', 'Column Pair Trends'],
+            'Score': [0.750000, 0.5],
+        }))
+
+    def test_report_end_to_end(self):
```
Comment (amontanez24) [inline tests/integration/reports/single_table/test_single_table_quality_report.py:63]: missing docstring

File: tests/unit/reports/single_table/test_single_table_quality_report.py [hunk 6]
```diff
@@ -1,455 +1,223 @@
...
+        mock_validate.assert_called_once_with(real_data, synthetic_data, metadata)
+        quality_report._properties['Column Shapes'].get_score.assert_called_with(
+            real_data, synthetic_data, metadata
+        )
+        quality_report._properties['Column Pair Trends'].get_score.assert_called_with(
+            real_data, synthetic_data, metadata
+        )
 
-        Expect that the overall quality score is returned.
+    def test__validate_property_generation(self):
```
Comment (amontanez24) [inline tests/unit/reports/single_table/test_single_table_quality_report.py:197]: If you update the method name update it here too please

**Review by amontanez24 — COMMENTED** [review]
Mostly minor comments. This is looking great! Things are really coming together

**Comment by R-Palazzo** [comment]
Thanks for the review @amontanez24. I addressed the comments in [eb364fe](https://github.com/sdv-dev/SDMetrics/pull/377/commits/eb364fee2d91b136254dd79b3b35d6e180d3f272)

**Review by fealho — APPROVED** [review]
💯

**Pull request merged**